flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -615,3 +615,17 @@ bool Fortran::lower::CharacterExprHelper::isArray(mlir::Type type) {
   }
   return false;
 }
+
+bool Fortran::lower::CharacterExprHelper::hasConstantLengthInType(
+    const fir::ExtendedValue &exv) {
+  auto type = fir::getBase(exv).getType();
+  if (auto refType = type.dyn_cast<fir::ReferenceType>())
+    type = refType.getEleTy();
```

Comment:
Shouldn't this use `fir::dyn_cast_ptrEleTy()` to be agnostic about the origin of the memory the CHARACTER data lives in?